hexrd/ui/import_data_panel.py

```diff
@@ -0,0 +1,247 @@
+import os
+import numpy as np
+
+from PySide2.QtCore import QObject, Signal
+from PySide2.QtWidgets import QFileDialog, QMessageBox
+
+from hexrd.ui.hexrd_config import HexrdConfig
+from hexrd.ui.image_file_manager import ImageFileManager
+from hexrd.ui.image_load_manager import ImageLoadManager
+from hexrd.ui.interactive_template import InteractiveTemplate
+from hexrd.ui.load_images_dialog import LoadImagesDialog
+from hexrd.ui import resource_loader
+from hexrd.ui.ui_loader import UiLoader
+
+import hexrd.ui.resources.calibration
+
+
+class ImportDataPanel(QObject):
+
+    # Emitted when new config is loaded
+    new_config_loaded = Signal()
+
+    def __init__(self, parent=None):
+        super(ImportDataPanel, self).__init__(parent)
+
+        loader = UiLoader()
+        self.ui = loader.load_file('import_data_panel.ui', parent)
+        self.it = None
+        self.edited_images = {}
+        self.completed_detectors = []
+
+        self.setup_connections()
+
+    def setup_connections(self):
+        self.ui.instruments.currentIndexChanged.connect(
+            self.instrument_selected)
+        self.ui.detectors.currentIndexChanged.connect(self.detector_selected)
+        self.ui.load.clicked.connect(self.load_images)
+        self.ui.add_template.clicked.connect(self.add_template)
+        self.ui.trans.clicked.connect(self.setup_translate)
+        self.ui.rotate.clicked.connect(self.setup_rotate)
+        self.ui.add_transform.clicked.connect(self.add_transform)
+        self.ui.button_box.accepted.connect(self.crop_and_mask)
+        self.ui.button_box.rejected.connect(self.clear)
+        self.ui.save.clicked.connect(self.save_file)
+        self.ui.complete.clicked.connect(self.completed)
+
+    def instrument_selected(self, idx):
+        if idx == 3:
+            self.ui.detectors.setCurrentIndex(0)
+            self.ui.detectors.setDisabled(True)
+        else:
+            instruments = ['TARDIS', 'PXRDIP', 'BBXRD']
+            det_list = self.get_instrument_detectors(instruments[idx])
+            if len(det_list) > 1:
+                self.load_instrument_config(instruments[idx])
+                self.new_config_loaded.emit()
+                self.ui.detectors.clear()
+                self.ui.detectors.insertItems(0, det_list)
+                self.ui.detector_label.setEnabled(True)
+                self.ui.detectors.setEnabled(True)
+
+    def get_instrument_detectors(self, instrument):
+        self.mod = resource_loader.import_dynamic_module(
+            'hexrd.ui.resources.templates.' + instrument)
+        contents = resource_loader.module_contents(self.mod)
+        dets = ['None']
+        for content in contents:
+            if isinstance(content, str) and not content.startswith('__'):
```

Comment:
What other type are we dealing with here?

Reply:
I believe that I originally added the check because `importlib.resources.contents` can also return directory objects, but since there would be no reason for sub-directories to be present in the templates directories this check can be removed.